lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet.py

```diff
@@ -0,0 +1,303 @@
+        return None
+
+    def get_by_name(self, droplet_name):
+        if not droplet_name:
+            return None
+        response = self.rest.get('droplets')
+        json_data = response.json
+        if response.status_code == 200:
+            for droplet in json_data['droplets']:
+                if droplet['name'] == droplet_name:
+                    return {'droplet': droplet}
+        return None
+
+    def get_droplet(self):
+        json_data = self.get_by_id(self.module.params['id'])
+        if not json_data and self.unique_name:
+            json_data = self.get_by_name(self.module.params['name'])
+        return json_data
+
+    def create(self):
+        json_data = self.get_droplet()
+        if json_data:
+            self.module.exit_json(changed=False, data=json_data)
+        response = self.rest.post('droplets', data=self.module.params)
+        json_data = response.json
+        if self.wait:
+            json_data = self.ensure_power_on(json_data['droplet']['id'])
+        self.module.exit_json(changed=True, data=json_data)
+
+    def delete(self):
+        json_data = self.get_droplet()
+        if json_data:
+            response = self.rest.delete('droplets/{0}'.format(json_data['droplet']['id']))
+            json_data = response.json
+            if response.status_code == 204:
+                self.module.exit_json(changed=True, msg='Droplet deleted')
+            self.module.fail_json(changed=False, msg='Failed to delete droplet')
+        else:
+            self.module.exit_json(changed=False, msg='Droplet not found')
+
+    def ensure_power_on(self, droplet_id):
+        end_time = time.time() + self.wait_timeout
+        while time.time() < end_time:
+            response = self.rest.get('droplets/{0}'.format(droplet_id))
+            json_data = response.json
+            if json_data['droplet']['status'] == 'active':
+                return json_data
+            time.sleep(min(2, end_time - time.time()))
+        self.module.fail_json(msg='Wait for droplet powering on timeout')
+
+
+def core(module):
+    state = module.params.pop('state')
+    droplet = DODroplet(module)
+    if state == 'present':
+        droplet.create()
+    elif state == 'absent':
+        droplet.delete()
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(choices=['present', 'absent'], default='present'),
+            oauth_token=dict(
+                aliases=['API_TOKEN'],
+                no_log=True,
+                fallback=(env_fallback, ['DO_API_TOKEN', 'DO_API_KEY'])
+            ),
+            name=dict(type='str'),
+            size=dict(aliases=['size_id']),
+            image=dict(aliases=['image_id']),
+            region=dict(aliases=['region_id']),
+            ssh_keys=dict(type='list'),
+            private_networking=dict(type='bool', default='no'),
+            backups=dict(type='bool', default='no'),
+            id=dict(aliases=['droplet_id'], type='int'),
```

Comment:
Please add droplet_id to `aliases:` in `DOCUMENTATION`

http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block